CMakeLists.txt

```diff
@@ -20,6 +20,7 @@ project(
 )
 
 set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)

```

Comment:
this is purely an improvement for vscode performance issues. Has no effect on builds.